python-sdk/pachyderm_sdk/api/pps/extension.py

```diff
@@ -148,3 +155,58 @@ def create_secret(
         ).encode()
 
         return super().create_secret(file=file)
+
+    def generate_datum(self, input_spec: "Input", batch_size: int) -> Iterator["DatumInfo"]:
```

Comment:
A bit pedantic, but I'd change the return type to `Generator[List["DatumInfo"], int, None]` which means "this is a generator object that yields lists of DatumInfo objects, can be sent integers, and returns nothing (None)